robottelo/cli/factory.py

```diff
@@ -1638,22 +1639,23 @@ def activationkey_add_subscription_to_repo(options=None):
             'subscription.'
         )
     # List the subscriptions in given org
-    result = Subscription.list(
+    subscriptions = Subscription.list(
         {u'organization-id': options['organization-id']},
-        per_page=False
+        per_page=False,
     )
     # Add subscription to activation-key
-    for subscription in result.stdout:
+    for subscription in subscriptions:
         if subscription['name'] == options['subscription']:
             if int(subscription['quantity']) == 0:
                 raise CLIFactoryError(
                     'All the subscriptions are already consumed')
-            result = ActivationKey.add_subscription({
-                u'id': options['activationkey-id'],
-                u'subscription-id': subscription['id'],
-                u'quantity': 1,
-            })
-            if result.return_code != 0:
+            try:
+                ActivationKey.add_subscription({
+                    u'id': options['activationkey-id'],
+                    u'subscription-id': subscription['id'],
+                    u'quantity': 1,
+                })
+            except CLICommandExecutionError:
```

Comment:
How about including this exception in the inner exception?

How about including the exception in this error message?

```python
except CLICommandExecutionError as err:
    raise ...  # include err in new exception
```